sudachi/src/config.rs

```diff
@@ -207,52 +288,82 @@ impl Config {
+    pub fn resolve_paths(&self, mut path: String) -> Vec<String> {
         if path.starts_with("$exe") {
             path.replace_range(0..4, &CURRENT_EXE_DIR);
+
+            let mut path2 = path.clone();
+            path2.insert_str(CURRENT_EXE_DIR.len(), "/deps");
+            return vec![path2, path];
         }
 
-        if path.starts_with("$cfg") {
-            let cfg_path = self.resource_dir.to_str().unwrap();
-            path.replace_range(0..4, cfg_path);
+        if path.starts_with("$cfg/") || path.starts_with("$cfg\\") {
+            let roots = self.resolver.roots();
+            let mut result = Vec::with_capacity(roots.len());
+            path.replace_range(0..5, "");
+            for root in roots {
+                let subpath = root.join(&path);
+                result.push(subpath.to_string_lossy().into_owned());
+            }
+            return result;
         }
 
-        path
+        vec![path]
     }
 
-    pub fn resolve_plugin_paths(&self, mut path: String) -> Vec<String> {
-        if path.starts_with("$exe") {
-            path.replace_range(0..4, &CURRENT_EXE_DIR);
+    /// Resolves given path to a path relative to anchors if its relative
+    pub fn complete_path<P: AsRef<Path> + Into<PathBuf>>(
+        &self,
+        file_path: P,
+    ) -> Result<PathBuf, ConfigError> {
+        let pref = file_path.as_ref();
+        // 1. absolute paths are not normalized
+        if pref.is_absolute() {
+            return Ok(file_path.into());
+        }
 
-            let mut path2 = path.clone();
-            path2.insert_str(CURRENT_EXE_DIR.len(), "/deps");
-            return vec![path2, path];
+        // 2. try to resolve paths wrt anchors
+        if let Some(p) = self.resolver.first_existing(pref) {
+            return Ok(p);
         }
 
-        if path.starts_with("$cfg") {
-            let cfg_path = self.resource_dir.to_str().unwrap();
-            path.replace_range(0..4, cfg_path);
+        // 3. try to resolve path wrt CWD
+        if pref.exists() {
+            return Ok(file_path.into());
         }
 
-        vec![path]
+        // Report an error
+        return Err(self.resolver.resolution_failure(&file_path));
     }
 
-    /// Resolves given path to a path relative to resource_dir if its relative
-    pub fn complete_path(&self, file_path: PathBuf) -> PathBuf {
-        Config::join_if_relative(&self.resource_dir, file_path)
-    }
+    pub fn resolved_system_dict(&self) -> Result<PathBuf, ConfigError> {
+        let sdic = self.system_dict.as_ref();
+        if sdic.is_none() {
+            return Err(ConfigError::MissingArgument("systemDict".to_owned()));
+        }
+
+        let sdic = sdic.unwrap();
 
-    fn join_if_relative(resource_dir: &PathBuf, file_path: PathBuf) -> PathBuf {
-        if file_path.is_absolute() {
-            file_path
-        } else {
-            resource_dir.join(&file_path)
+        if sdic.is_absolute() {
+            return Ok(sdic.clone());
         }
```

Comment:
`complete_path(sdic)` seems doing the same check.

Reply:
Nice catch!